packages/node/src/main.ts

```diff
@@ -11,8 +30,15 @@ export function main<T>(operation: Operation<T>): Context<T> {
         process.on('SIGUSR1', debug);
         return yield operation;
       } catch(e) {
-        console.error(e);
-        process.exit(-1);
+        if(e.name === 'EffectionMainError') {
+          if(!e.silent) {
```

Comment:
I don't know if this is your experience, but mine is that most of the time that I go to the effort of throwing a custom error designed to kill the process, I'm going to make the output custom as well, so what do you think about making `silent` true by default?

In other words, usually what you want is to either a) write a message to stderr (no stack trace), or b) log it fully as an uncaught error. (b) seems to correspond to the case where where you don't have a `MainError`.

Having a boolean option to either do nothing or fully log the error using the default `console.error`, doesn't seem flexible enough.

In other words, let's say we wanted to do this:

```
$ bigtest server --launch waterfox
❗ can't find a driver for 'waterfox'. Did you mean 'firefox'?
```

What if we  _always_ log the `message` of a `MainError` if it's defined (we can make it undefined by default), and then do our `console.error` if there is a `logBacktrace` option.

then again, whether to log the backtrace doesn't feel like something that should be controlled by the site where the exception is thrown, but rather from where it's caught via a `LOG_LEVEL` or something.

Reply:
This is a great suggestion :) I don't really think it's that useful to be able to print the stacktrace at all though. I did leave an option for it in there, just in case.